src/App.tsx

```diff
@@ -1,25 +1,44 @@
-import React from 'react';
+import { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const startIndex = (currentPage - 1) * itemsPerPage + 1;
+  const endIndex = Math.min(startIndex + itemsPerPage - 1, 42);
```

Comment:
The calculation of `startIndex` and `endIndex` is slightly off. The `startIndex` should be `(currentPage - 1) * itemsPerPage`, not `(currentPage - 1) * itemsPerPage + 1`, to correctly reflect the zero-based index of the items array. Similarly, `endIndex` should be `Math.min(startIndex + itemsPerPage, items.length)` to ensure it does not exceed the total number of items.